### PR TITLE
Implement a single-file upload widget.

### PR DESCRIPTION
**Pull request by dawehner**
## Testing instructioins

* Checkout the branch
* Import the ```config/install/rest.resource.file.upload.yml```
* Create a recipe
* Try to upload the file

## Screenshot

react_app

### CODE REVIEW

File: src/components/02_atoms/FileUpload/FileUpload.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  state = {
+    drag: false,
+    file: null,
+  };
+
+  onDragOver = event => {
+    this.setState({
+      drag: true,
+    });
```
Comment (mattgrill) [inline src/components/02_atoms/FileUpload/FileUpload.js]: Do we want to make sure to `preventDefault` and `stopPropagation` happen after?

File: src/components/02_atoms/FileUpload/FileUpload.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        this.props.onFileUpload(createdFile);
+        this.setState({
+          drag: false,
+          file: createdFile,
+        });
+      }
+    };
+
+    reader.readAsArrayBuffer(files[0]);
```
Comment (mattgrill) [inline src/components/02_atoms/FileUpload/FileUpload.js]: Should we add a todo to handle multi-file uploads.

File: src/components/02_atoms/FileUpload/FileUpload.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+  render() {
+    return (
+      <Element
+        drag={this.state.drag}
+        onDragOver={this.onDragOver}
+        onDragLeave={this.onDragLeave}
+        onDrop={this.onDrop}
+      >
+        {(this.state.file && this.state.file.filename[0].value) ||
```
Comment (mattgrill) [inline src/components/02_atoms/FileUpload/FileUpload.js]: Should we check length here, not just that the key exists?

**Comment by mattgrill** [comment]
Two follow up issues for this should be created, 

- handling multiple files
- some styles.

**Comment by justafish** [comment]
I tested this, put the image uploaded unfortunately has 0 bytes

screenshot from 2018-07-08 21-11-35

**Comment by mattgrill** [comment]
files___drupal

actually creating the files now.

**Pull request merged**